plugins/logs/plugin.go

```diff
@@ -402,11 +402,8 @@ func (p *Plugin) Start(ctx context.Context) error {
 func (p *Plugin) Stop(ctx context.Context) {
 	p.logInfo("Stopping decision logger.")
 
-	gracefulDeadline, _ := ctx.Deadline()
-	gracefulShutdownPeriod := gracefulDeadline.Sub(time.Now())
-
-	if p.config.Service != "" && gracefulShutdownPeriod > 0 {
-		p.flushDecisions(context.WithTimeout(ctx, gracefulShutdownPeriod))
+	if _, ok := ctx.Deadline(); ok && p.config.Service != "" {
```

Comment:
Nit: seems like we only flush decisions if the built-in HTTP logger is used. What about custom loggers? They require a signal as well. Not something we need to deal with here though.